### PR TITLE
Todo app

### PR DESCRIPTION
**Pull request by mntedev**
I have completed the following tasks.

    implemented add todo, delete todo, update todo and render todos using class components.
    implemented add todo, delete todo, update todo and render todos using functional components.
    used useState and UseEffect hooks
    added routing using react-router dom
    added styling for react components
    added icon to react components

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,56 @@
-# React-Todo-App
\ No newline at end of file
+![](https://img.shields.io/badge/Microverse-blueviolet)
+
+# React-Todo-App
+
+> This project is created to demonistrate how to create react application. To learn react class compoent, events, fucntional componets and hooks.
```
Comment (waqaskanju) [inline README.md]: change spelling

Comment (waqaskanju) [reply README.md]: Change spelling of component and functional

File: README.md [hunk 2]
```diff
@@ -1 +1,56 @@
...
+
+Feel free to check the [issues page](../../issues/).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (waqaskanju) [inline README.md:56]: Mit.md file not found

File: public/index.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
+    />
+    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
```
Comment (waqaskanju) [inline public/index.html:12]: Remove files which you are not using.
image

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/*eslint-disable */
+import ReactDOM from 'react-dom/client';
+import React from 'react';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+import TodoContainer from './functionBased/components/TodoContainer';
+import About from './functionBased/pages/About';
+import NotMatch from './functionBased/pages/NotMatch';
+import Navbar from './functionBased/components/Navbar';
```
Comment (waqaskanju) [inline src/index.js:8]: Correct the nav bar it should be only visible in Mobile view not in desktop.
image

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+  <React.StrictMode>
+    <Router>
+      <Navbar />
+      <Routes>
+        <Route exact path="/" element={<TodoContainer />} />
+        <Route path="/about" element={<About />} />
```
Comment (waqaskanju) [inline src/index.js:16]: You class base show errors.
image

File: src/classBased/components/TodoList.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  render() {
+    return (
+      <ul>
+        {this.props.todos.map((todo) => (
+          <TodoItem
+            setUpdate={this.props.setUpdate}
+            key={todo.id}
+            todo={todo}
+            handleChangeProps={this.props.handleChangeProps}
+            deleteItem={this.props.deleteItem}
```
Comment (waqaskanju) [inline src/classBased/components/TodoList.js:14]: The delete option in class base is not working.
image

File: src/classBased/components/TodoItem.module.css [hunk 7]
```diff
@@ -0,0 +1,29 @@
+.item {
+  font-size: 1.2rem;
+  list-style-type: none;
+  padding: 17px 0;
+  border-bottom: 1px solid #eaeaea;
+}
```
Comment (waqaskanju) [inline src/classBased/components/TodoItem.module.css:6]: The rest of application is working great.

**Review by waqaskanju — CHANGES_REQUESTED** [review]

**Comment by mntedev** [comment]
the menu is shown on both mobile and desktop devices, and can you give me a suggestion on how to correct the license.

**Review by waqaskanju — APPROVED** [review]

**Comment by waqaskanju** [comment]
You code is working great

**Comment by waqaskanju** [comment]
> I have completed the following tasks.
> 
> ```
> implemented add todo, delete todo, update todo and render todos using class components.
> implemented add todo, delete todo, update todo and render todos using functional components.
> used useState and UseEffect hooks
> added routing using react-router dom
> added styling for react components
> added icon to react components
> ```

Good Work Worish!
Your readme File is good
There are no linter errors
Your project is runnable.

There are some suggestion listed below.

**Pull request merged**